internal/pkg/api/repository/result/interface.go

```diff
@@ -42,6 +42,8 @@ type Repository interface {
 	GetExecutions(ctx context.Context, filter Filter) ([]testkube.Execution, error)
 	// GetExecutionTotals gets the statistics on number of executions using a filter, but without paging
 	GetExecutionTotals(ctx context.Context, paging bool, filter ...Filter) (result testkube.ExecutionsTotals, err error)
+	// GetNextExecutionNumber gets next execution number by test name
+	GetNextExecutionNumber(ctx context.Context, testName string) (number int, err error)
```

Comment:
I would close it in separate interface and add it here 

Additionally move out logic to separate structure to allow easy composition